includes/admin/views/html-admin-page-addons.php

```diff
@@ -113,21 +147,53 @@ class="<?php echo $current_section === $section->slug ? 'current' : ''; ?>"
 						?>
 						<li class="product">
 							<div class="product-details">
-								<?php if ( ! empty( $addon->image ) ) : ?>
-									<span class="product-img-wrap"><img src="<?php echo esc_url( $addon->image ); ?>" /></span>
+								<?php if ( ! empty( $addon->image ) || ! empty( $addon->icon ) ) : ?>
+									<span class="product-img-wrap">
+										<?php /* Show an icon if it exists, and an image otherwise */ ?>
+										<img src="<?php echo esc_url( empty( $addon->icon ) ? $addon->image : $addon->icon ); ?>" />
```

Comment:
I'm thinking we shouldn't show the image here if there's no icon: it's not designed for this scale, and won't always look good; and it'll hopefully encourage extension developers to add an app icon to their extensions.

Reply:
Indeed, this way we'd encourage vendors to use icons. Applied in https://github.com/woocommerce/woocommerce/pull/30840/commits/2461d20e2d51482a2ed83c3b97d9a44f8c0479cb.